Sources/Schema.swift

```diff
@@ -100,7 +101,27 @@ extension SchemaBuilder {
         }
 
         let name = components[2]
+
+        if let cachedSchema = ReferenceResolver.cachedSchemaReferences[name] {
+            return Structure(name: name, structure: cachedSchema)
+        }
+        if let resolving = ReferenceResolver.resolvingReferences[name], resolving {
+            return Structure(name: name, structure: .unresolvedReference(pointer.path))
```

Comment:
Unless I'm misunderstanding, if *any* reference is unresolved, the spec should be invalid and fail to map. Could you give an example of a case where an unresolved reference should be a valid schema? What does that mean to a consumer of the library when they're trying to produce a strictly typed model?